### ISSUE TITLE
Disconnect peers when node can't advance

### ISSUE THREAD

**Issue by dangershony** [stratisproject/StratisBitcoinFullNode#2146]
Create an additional behaviour to support this functionality.
Possibly put this in `PosConsensusManagerBehaviour`

If a node is connected to the max outbounds limit, and all the nodes are not PH enabled or whitelisted we need to disconnect some nodes to try to make space for PH enabled nodes or our, otherwise we will not be able to advance our our consensus height.

**Comment by dangershony** [comment]
A suggestion is that 50% of slots for outbound are saved for PH nodes.
We can also boost the number of maximum outbound nodes.

**Comment by MithrilMan** [comment]
so do you want a settings like "phConnectionReserved" and disconnect any nodes that doesn't support PH, trying to occupy that slots?

**Comment by MithrilMan** [comment]
I'll create a new behavior because I see PosConsensusManagerBehaviour is not (yet?) available

**Comment by noescape00** [comment]
ProvenHeadersReservedSlotsBehavior (PHRSB)



Behavior should be subscribed to version message.

When new peers connects and starts handshaking (sends us version payload) we estimate if peer supports PH or not based on the version provided. 

If peer supports PH we check how many outbound connections we have. If we have all outbound slots taken we will disconnect single non-PH peer. 

If peer doesn't support PH and all outbound slots are taken we will drop the connection with this peer. 



In both cases if after this peer connects to us we still have at least one outbound slot we don't do anything, without checking if peer is PH or not.



Basically the idea is to have single slot reserved at all times except for cases when we have all our outbound connections to PH nodes.

**Comment by MithrilMan** [comment]
ok so no more slots at all, makes sense and doesn't involve many external components

**Comment by dangershony** [comment]
I think we also need to be aware that we do this check only for outbound connections (I think there is a flag in the behaviour or the node to check that), so check that the incoming connection is an outbound we initiated and its the last node in N allowed outbounds
